main.py

```diff
@@ -1,129 +1,171 @@
+from datetime import datetime, timedelta
 import requests
 import os
 from dotenv import load_dotenv
 from flask import Flask, jsonify
-from datetime import datetime
 from sql_client import SQLClient
 
 load_dotenv()
 
+api_key = os.getenv('WEATHER_API_KEY')
+api_url_base = os.getenv('WEATHER_API_URL', 'https://dataservice.accuweather.com')
 ipinfo_api_token = os.getenv('IPINFOAPITOKEN')
 
 app = Flask(__name__)
 
 db_client = SQLClient()
 
+# Simple cache to store weather data
+weather_cache = {}
+
+
 def create_weather_table():
     create_table_query = """
     CREATE TABLE IF NOT EXISTS weather (
         id INT AUTO_INCREMENT PRIMARY KEY,
-        bikeable BOOLEAN,
         icon VARCHAR(255),
-        lat FLOAT,
-        lng FLOAT,
+        lat VARCHAR(255),
```

Comment:
Why not use a float?

Reply:
because for some kind of reason the query will break when using float for this value

Reply:
Breaks the query